service/service.go

```diff
@@ -31,13 +31,47 @@ import (
 	"github.com/vulcand/vulcand/supervisor"
 )
 
+type ControlCode int
+
+const (
+	ControlCodeGracefulShutdown ControlCode = iota
+	ControlCodeImmediateShutdown
+	ControlCodeForkChild
+)
+
+func waitForSignals() chan ControlCode {
+	sigC := make(chan os.Signal, 1024)
+	signal.Notify(sigC, os.Interrupt, os.Kill, syscall.SIGTERM, syscall.SIGUSR2)
+	controlC := make(chan ControlCode, 1024)
+
+	go func() {
+		for {
+			signal := <-sigC
+			log.Infof("Got signal '%s'", signal)
+
+			switch signal {
+			case syscall.SIGTERM, syscall.SIGINT:
+				controlC <- ControlCodeGracefulShutdown
+			case syscall.SIGKILL:
+				controlC <- ControlCodeImmediateShutdown
+			case syscall.SIGUSR2:
```

Comment:
Good catch, this case had a typo: it was sending `ControlCodeGracefulShutdown` wghen it should have been sending `ControlCodeForkChild`. Fixed in aa74c54.